src/mavsdk/core/callback_list.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <utility>
+#include <vector>
+
+namespace mavsdk {
+
+template<typename... Args> class CallbackList;
+
+template<typename... Args> class Handle {
+public:
+    explicit Handle(uint64_t id) : _id(id) {}
+
+private:
+    uint64_t _id;
+
+    friend CallbackList<Args...>;
+};
+
+template<typename... Args> class CallbackList {
+public:
+    Handle<Args...> subscribe(const std::function<void(Args...)>&& callback)
```

Comment:
Is it necessary to already fix the return type?

Reply:
Also, would you be able to copy the callbacklist and handle?

Reply:
The return type is to have better type safety, but we could just use generic handles instead. Depends what we prefer.